Fix conversation list: remove GitHub link and show created_at date

rbren opened the pull request:
This PR implements two fixes for the conversation list:

1. Removes the link to GitHub when a repo is attached (changes the button to a plain text span)
2. Shows the `created_at` date rather than `updated_at` in the conversation card

The changes are backward compatible, with `createdAt` being an optional prop that falls back to `lastUpdatedAt` if not provided.

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:e0571b3-nikolaik   --name openhands-app-e0571b3   docker.all-hands.dev/all-hands-ai/openhands:e0571b3
```

enyst commented on frontend/src/components/features/conversation-panel/conversation-card.tsx:
```diff
@@ -205,7 +209,9 @@ export function ConversationCard({
           <p className="text-xs text-neutral-400">
-            <time>{formatTimeDelta(new Date(lastUpdatedAt))} ago</time>
+            <time>
+              {formatTimeDelta(new Date(createdAt || lastUpdatedAt))} ago
```
Why not the other way around?

I find lastUpdatedAt useful to figure out which conversation is which. 😅

createdAt doesn't seem too useful? Who is going to remember it and why?

rbren replied:
Maybe this is personal preference, but the list is sorted by createdAt, so it looks weird to me when the times show updatedAt. Everything looks out of order

We could also sort by updatedAt, but if you have two active conversations they're going to flip back and forth in slots 1 and 2, which would be a bad experience

enyst replied:
Afaik the conversations in the panel are ordered by lastUpdatedAt. So it seems to make sense to keep that visible, rather than the one doesn't matter.

Visually it would be strange to see conversations switch order for no visible reason. Maybe I'm missing something?

enyst replied:
Oh, I was just answering. I thought they're ordered by updatedAt, hmm

enyst replied:
It looks like you're right. There are some in my list "out of order" with what they display.

enyst replied:
Actually... I'm not sure that's right: 
> if you have two active conversations they're going to flip back and forth in slots 1 and 2, which would be a bad experience

UpdatedAt is a bad name, I remember it means ... something like last time the session was started? So they don't, and wouldn't, switch order (much)

rbren replied:
<img width="407" alt="Screenshot 2025-03-22 at 6 59 28 PM" src="https://github.com/user-attachments/assets/5f987c63-1187-4869-b7b4-e2247b4cbd1a" />
how about this?

enyst replied:
I think if updatedAt is the last time you opened the conversation, it makes sense to me to be what they're sorted on, and displayed. It won't flip back and forth for active conversations, their last session started when it started.

I could go with both if you want.

enyst replied:
Currently:

<img width="257" alt="image" src="https://github.com/user-attachments/assets/958537fc-9879-4522-a7b6-cc86d3bf9e44" />

enyst replied:
This sets updatedAt:
https://github.com/All-Hands-AI/OpenHands/blob/306188817f10c4e63f25c60d4cfaa92e18354aa6/openhands/server/conversation_manager/standalone_conversation_manager.py#L292

enyst reviewed (approved):
OK so what I was thinking here is no good, because it would actually flip all the time between active conversations.

I think maybe it's still worth displaying somehow what the user will remember - IMHO that's last time they worked on something, more likely than when they first worked on something. 🤔

Or maybe it won't matter anymore, because titles will do.

tofarr reviewed (changes requested):
We probably need to remove those console.log statements, and will there be enough space on smaller screens - but aside from that LGTM.

rbren commented:
@openhands please make sure lint, build, and test all pass

rbren commented:
I have fixed the issues in the PR:

1. Fixed lint errors by removing console.log statements
2. Fixed build errors by handling the case where createdAt is undefined
3. Updated tests to match the new time format

All lint, build, and test checks are now passing.

tofarr reviewed (approved):
🍰
